src/Functions/dateDiff.cpp

```diff
@@ -473,21 +473,158 @@ REGISTER_FUNCTION(DateDiff)
+
+For an alternative to [`date_diff`](#date_diff), see function [`age`](#age).
+    )";
+    FunctionDocumentation::Syntax syntax_date_diff = R"(
+date_diff(unit, startdate, enddate, [timezone])
+    )";
+    FunctionDocumentation::Arguments arguments_date_diff =
+    {
+        {"unit", R"(The type of interval for result.
+
+| Unit | Possible values |
+|------|----------------|
+| nanosecond | `nanosecond`, `nanoseconds`, `ns` |
+| microsecond | `microsecond`, `microseconds`, `us`, `u` |
+| millisecond | `millisecond`, `milliseconds`, `ms` |
+| second | `second`, `seconds`, `ss`, `s` |
+| minute | `minute`, `minutes`, `mi`, `n` |
+| hour | `hour`, `hours`, `hh`, `h` |
+| day | `day`, `days`, `dd`, `d` |
+| week | `week`, `weeks`, `wk`, `ww` |
+| month | `month`, `months`, `mm`, `m` |
+| quarter | `quarter`, `quarters`, `qq`, `q` |
+| year | `year`, `years`, `yyyy`, `yy` |
+)"},
+        {"startdate", "The first time value to subtract (the subtrahend). [`Date`](../data-types/date.md) or [`Date32`](../data-types/date32.md) or [`DateTime`](../data-types/datetime.md) or [`DateTime64`](../data-types/datetime64.md)."},
+        {"enddate", "The second time value to subtract from (the minuend). [`Date`](../data-types/date.md) or [`Date32`](../data-types/date32.md) or [`DateTime`](../data-types/datetime.md) or [`DateTime64`](../data-types/datetime64.md)."},
+        {"timezone", "Optional. Timezone name. If specified, it is applied to both `startdate` and `enddate`. If not specified, timezones of `startdate` and `enddate` are used. If they are not the same, the result is unspecified. [`String`](../data-types/string.md)."}
+    };
+    FunctionDocumentation::ReturnedValue returned_value_date_diff = "Returns the difference between `enddate` and `startdate` expressed in `unit`. [`Int64`](../data-types/int-uint.md).";
+    FunctionDocumentation::Examples examples_date_diff =
+    {
+        {"Calculate date difference in hours", R"(
+SELECT dateDiff('hour', toDateTime('2018-01-01 22:00:00'), toDateTime('2018-01-02 23:00:00')) AS res
+        )",
+        R"(
+┌─res─┐
+│  25 │
+└─────┘
+        )"},
+        {"Calculate date difference in different units", R"(
+SELECT
+    toDate('2022-01-01') AS e,
+    toDate('2021-12-29') AS s,
+    dateDiff('day', s, e) AS day_diff,
+    dateDiff('month', s, e) AS month_diff,
+    dateDiff('year', s, e) AS year_diff
+        )",
+        R"(
+┌──────────e─┬──────────s─┬─day_diff─┬─month_diff─┬─year_diff─┐
+│ 2022-01-01 │ 2021-12-29 │        3 │          1 │         1 │
+└────────────┴────────────┴──────────┴────────────┴───────────┘
+        )"}
+    };
+    FunctionDocumentation::IntroducedIn introduced_in_date_diff = {23, 4};
+    FunctionDocumentation::Category category_date_diff = FunctionDocumentation::Category::DateAndTime;
+    FunctionDocumentation documentation_date_diff =
+    {
+        description_date_diff,
```

Comment:
\- suffix `_date_diff` (I think that would work as `dateDiff` and `age` are defined in different C++ scopes)